python/tests/rest_api/test_rest_api.py

```diff
@@ -1,95 +1,283 @@
+)
+from rest_api.models.definition_model import DefinitionModel, to_definition_class, to_definition_model
+from rest_api.models.file_model import FileModel, FilePathModel, FilePathRenameModel, to_file_model
+from rest_api.aac_rest_app import app, refresh_available_files_in_workspace, _get_available_files_in_workspace
+
+class TestRestApiCommands(TestCase):
+    test_client = TestClient(app)
+
+    def test_get_available_commands(self):
+        response = self.test_client.get("/commands")
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+
+    def test_execute_check_command(self):
+        command_name = "check"
+        test_model = parse(TEST_MODEL)[0]
+
+        request_arguments = CommandRequestModel(name=command_name, arguments=[TEST_MODEL, "False", "False"])
+        response = self.test_client.post("/command", data=json.dumps(jsonable_encoder(request_arguments)))
+
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertTrue(response.json().get("success"))
+        self.assertIn("success", response.text)
+        self.assertIn(command_name, response.text)
+        self.assertIn(test_model.name, response.text)
+
+    def test_execute_gen_plugin_command(self):
+        command_name = "gen-plugin"
+        test_model = parse(TEST_MODEL)[0]
+
+        with TemporaryDirectory() as temp_dir:
+            request_arguments = CommandRequestModel(name=command_name, arguments=[TEST_MODEL, temp_dir, temp_dir, temp_dir, "True", "True", "False"])
+            response = self.test_client.post("/command", data=json.dumps(jsonable_encoder(request_arguments)))
+
+            self.assertEqual(HTTPStatus.OK, response.status_code)
+            self.assertTrue(response.json().get("success"))
+            self.assertIn("success", response.text)
+            self.assertIn(command_name, response.text)
+
+    def test_execute_check_command_fails(self):
+        command_name = "check"
+        with self.assertRaises(Exception) as context:
+            request_arguments = CommandRequestModel(name=command_name, arguments=[BAD_TEST_MODEL, "False", "False"])
+            response = self.test_client.post("/command", data=json.dumps(jsonable_encoder(request_arguments)))
+
+
+class TestAacRestApiFiles(TestCase):
+    test_client = TestClient(app)
+
+    def test_post_and_get_files(self):
+        filepath = "tests/calc/model/calculator.yaml"
+        self.assertTrue(os.path.isfile(filepath))
+
+        file_model = [FilePathModel(uri=os.path.abspath(filepath))]
+        self.test_client.post("/files/import", data=json.dumps(jsonable_encoder(file_model)))
+        response = self.test_client.get("/files/context")
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertIn("calculator.yaml", response.text)
+
+        available_files = self.test_client.get("/files/available")
+        self.assertNotIn("calculator.yaml", available_files.text)
+
+    def test_get_available_aac_files(self):
```

Comment:
Yeah, I am stuck on this one. I've tried changing out the file used for the removal test in case it was an order of execution for the multiple test cases making it get completed prior to the test checking available when all are run. But that didn't solve it. Looked like you had started applying a refresh of the workspace at the start of all the file and definition tests, so implemented in the few that had it missing. But that didn't solve it either. Was checking the load to context places and they all seem to be working correctly. But looking at the context that gets dumped when the test fails, it seems to only be the core context. But I'm not finding anywhere in your tests or rest app logic that tells the context to filter to core.

Reply:
So it looks like the global context may be working TOO well, because the calculator file is still loaded into the context from the previous test, even when I refresh the app.

Reply:
Scratch that, that isn't what is happening either.

Reply:
It passes with pytest, but not with python -m unittest, so I am not sure what is going on.

Reply:
Yeah, I am thoroughly confused on this one as well. Interacting with the context and files provided by the Rest-API is fairly thoroughly tested with all the other tests contained in that Test Case class, so I think it would be okay if we drop that one for now since it seems to be we are misunderstanding the relationship between context and available files.